src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -83,13 +85,18 @@ private static boolean isValidPath(Path filePath) {
      *
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
+    public void save(AddressBook addressBook, boolean wasRunning) throws StorageOperationException, StorageFileDeletedException {
 
         /* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
          */
+        File storageFile = path.toFile();
+        if (wasRunning && !storageFile.exists()){
```

Comment:
- the check is correct
- this implementation stops the program flow and halts the program
 - better way is to let the program create the file as it was doing earlier, but use exception as a mechanism to let the user know of the fact that the file was deleted.